docs/model-adaptivity.md

```diff
@@ -0,0 +1,163 @@
+as long as the `switching_function` contains values other than 0.
+Model evaluation - in the call `solve_variant(micro_sims_input, dt)` - is delegated to the regular
+(non-model-adaptive) `micro_sim_solve(micro_sims_input, dt)` method.
+
+### Interfaces
+
+```python
+class MicroSimulation: # Name is fixed
+    def __init__(self, sim_id):
+        """
+        Constructor of class MicroSimulation.
+
+        Parameters
+        ----------
+        sim_id : int
+            ID of the simulation instance, that the Micro Manager has set for it.
+        """
+
+    def initialize(self) -> dict:
+        """
+        Initialize the micro simulation and return initial data which will be used in computing adaptivity before the first time step.
+
+        Defining this function is OPTIONAL.
+
+        Returns
+        -------
+        initial_data : dict
+            Dictionary with names of initial data as keys and the initial data itself as values.
+        """
+
+    def solve(self, macro_data: dict, dt: float) -> dict:
+        """
+        Solve one time step of the micro simulation for transient problems or solve until steady state for steady-state problems.
+
+        Parameters
+        ----------
+        macro_data : dict
+            Dictionary with names of macro data as keys and the data as values.
+        dt : float
+            Current time step size.
+
+        Returns
+        -------
+        micro_data : dict
+            Dictionary with names of micro data as keys and the updated micro data a values.
+        """
+
+    def set_state(self, state):
+        """
+        Set the state of the micro simulation.
+        """
+
+    def get_state(self):
+        """
+        Return the state of the micro simulation.
+        """
+
+    def output(self):
+        """
+        This function writes output of the micro simulation in some form.
+        It will be called with frequency set by configuration option `simulation_params: micro_output_n`
+        This function is *optional*.
+        """
+```
+
+For this the default MicroSimulation still serves as the model interface, while the `(set)|(get)_state()` methods
+are called to transfer internal model parameters from one to another.
+The list of provided models is interpreted in decreasing fidelity order. In other words, the first one
+is likely to be the full order model, while subsequent ones are ROMs.
+
+```python
+def switching_function(
+    resolutions: np.ndarray,
+    locations: np.ndarray,
+    t: float,
+    inputs: list[dict],
+    prev_output: dict,
+    active: np.ndarray,
+) -> np.ndarray:
+    """
+    Switching interface function, use as reference
+
+    Parameters
+    ----------
+    resolutions : np.array - shape(N,)
+        Array with resolution information as get_sim_class_resolution would return for a sim obj.
+    locations : np.array - shape(N,D)
+        Array with gaussian points for all sims. D is the mesh dimension.
```

Comment:
I am not sure about having macro locations as part of the switching function inputs. I could imagine the switching being solely based on the macro- and micro-scale data available at a particular point.